repos/system_upgrade/el7toel8/actors/postgresqlcheck/libraries/library.py

```diff
@@ -0,0 +1,78 @@
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import CalledProcessError, api, run
+
+def get_postgresql_packages():
+    """ Get list of installed PostgreSQL packages """
+
+    for rpm_pkgs in api.consume(InstalledRedHatSignedRPM):
+        return [pkg for pkg in rpm_pkgs.items if pkg.name.startswith('postgresql')]
+
+def get_postgresql_extensions():
+    """ Get list of PostgreSQL extensions """
+    result = []
+    try:
+        output = run([
+            '/usr/bin/su',
+            '-l', 'postgresql'
+            '-c', 'psql -tAc \'\dx\''
+        ], split=True)['stdout']
+    except CalledProcessError:
+        api.current_logger().debug('Unable to list PostgreSQL extensions: psql failed to run')
+        return False
```

Comment:
Wouldn't False here cause error later, when the output of this function is taken as array? Maybe returning an empty array in this case would be better.

Reply:
Its probably even better to return None as it’s not an empty list of extensions but the command failed

If you check if value instead of if len(value) then this will work with false none and an empty list

Reply:
I agree. Returning None seems much more universal as @vinzenz pointed out. Fixed.